fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -514,21 +514,40 @@ public JournalObservable getJournalObservable() {
         return journalObservable;
     }
 
+    public TNodesInfo createNodesInfo(boolean isExternalCluster, long warehouseId, SystemInfoService systemInfoService) {
+        TNodesInfo nodesInfo = new TNodesInfo();
+        if (isExternalCluster) {
```

Comment:
should not. External query needs to be warehouse awareness too so the resource for the query can be limited to a certain warehouse.

Reply:
ok, warehouse was purposely not considered.. as that would need more refactoring work, let me check it.